### PR TITLE
Switch to jansson sm Remove YAJL and Jansson

### PR DESCRIPTION
**Pull request by xw19**
Remove yajl and jansson as gitsubmodule

### CODE REVIEW

**Review by themoriarti — APPROVED** [review]
LGTM

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -35,6 +35,7 @@ jobs:
 
           setup: |
             git submodule update --init --recursive
+            git clone https://github.com/akheron/jansson
```
Comment (saschagrunert) [inline .github/workflows/test.yaml:38]: For what do we need this?

File: Containerfile [hunk 2]
```diff
@@ -12,12 +12,14 @@ RUN apt-get update && \
 
 COPY . libocispec
 
-RUN cd libocispec/jansson && autoreconf -fi && ./configure && make && make install
+RUN git clone https://github.com/akheron/jansson
+
+RUN cd jansson && autoreconf -fi && ./configure && make && make install
```
Comment (saschagrunert) [inline Containerfile:17]: Can we just install a package?

**Review by saschagrunert — COMMENTED** [review]
Just 2 nits, otherwise LGTM

File: Containerfile (hunk 2 above)
Comment (xw19) [reply Containerfile]: Fixed

**Comment by xw19** [comment]
@saschagrunert  PTAL

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -35,23 +35,13 @@ jobs:
 
           setup: |
             git submodule update --init --recursive
-            git clone https://github.com/akheron/jansson
 
           install: |
             apt-get update -q -y
-            apt-get install -q -y python3 automake libtool autotools-dev git make cmake pkg-config gcc wget xz-utils
+            apt-get install -q -y python3 automake libtool autotools-dev git make cmake pkg-config gcc wget xz-utils libjansson4 libjansson-dev
```
Comment (saschagrunert) [inline .github/workflows/test.yaml:41]:
```suggestion
            apt-get install -q -y python3 automake libtool autotools-dev git make cmake pkg-config gcc wget xz-utils libjansson-dev
```

File: Containerfile [hunk 4]
```diff
@@ -7,18 +7,14 @@ RUN apt-get update && \
         git \
         libtool \
         pkg-config \
-        python3
+        python3 \
+        libjansson4 \
```
Comment (saschagrunert) [inline Containerfile:11]:
```suggestion
```

**Review by saschagrunert — COMMENTED** [review]
The dev package should also provide the lib, right?

File: .github/workflows/test.yaml (hunk 3 above)
Comment (xw19) [reply .github/workflows/test.yaml]: fixed

Comment (saschagrunert) [reply .github/workflows/test.yaml]: I was wrong, it broke the CI

File: Containerfile (hunk 2 above)
Comment (themoriarti) [reply Containerfile]: Sure
```
apt install libyajl-dev
apt install libjansson-dev
```

**Pull request merged**